include/deal.II/trilinos/nox.h

```diff
@@ -0,0 +1,250 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2022 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_trilinos_nox
+#define dealii_trilinos_nox
+
+#include <deal.II/base/config.h>
+
+#ifdef DEAL_II_WITH_TRILINOS
```

Comment:
I don't think that this is enough. I think that you have to edit [`configure_20_trilinos.cmake`](https://github.com/dealii/dealii/blob/master/cmake/configure/configure_20_trilinos.cmake#L153) to add NOX as an optional dependency, and then use `#ifdef DEAL_II_TRILINOS_WITH_NOX` instead of this.